cardano-api/src/Cardano/Api/TxMetadata.hs

```diff
@@ -159,9 +158,20 @@ fromShelleyMetaData (Shelley.MetaData mdMap) =
 toShelleyMetadata :: TxMetadata -> Shelley.MetaData
 toShelleyMetadata (TxMetadataShelley m) = m
 
-toShelleyMetadataHash :: Ledger.Era ledgerera
+-- TODO: Could not deduce: Cardano.Ledger.Core.Metadata ledgerera ~ Shelley.MetaData
+--
+-- The ledger has turned Metadata in a type family.
+-- 'TxMetadata' is a newtype around 'Shelley.MetaData', which is what the type
+-- family instance Core.Metadata (ShelleyEra c) will reduce to.
+--
+-- The result type of this function is parametric in @ledgerera@ (btw, why not
+-- @ledgerEra@? @ledgerera@ is confusing to read and write), while the
+-- 'TxMetadata' is specific to the Shelley era. This can no longer remain the
+-- case. Either the result type must be fixed to the Shelley era or the
+-- 'TxMetadata' type must be generalised to any era (the right thing to do IMO).
+toShelleyMetadataHash :: () -- TODO Shelley.ValidateMetadata ledgerera
                       => TxMetadata -> Shelley.MetaDataHash ledgerera
-toShelleyMetadataHash (TxMetadataShelley m) = Shelley.hashMetaData m
+toShelleyMetadataHash = undefined -- (TxMetadataShelley m) = Shelley.hashMetadata m
```

Comment:
Can somebody from the node cli team take care of this? There are more type errors because of this change.